test/Microsoft.Framework.WebEncoders.Tests/HtmlEncoderTests.cs

```diff
@@ -0,0 +1,269 @@
+            Assert.Equal("&#xFFFF;", encoder.HtmlEncode("\uFFFF")); // we still always encode non-chars and other forbidden chars
+        }
+
+        [Fact]
+        public void Ctor_WithUnicodeBlocks()
+        {
+            // Arrange
+            HtmlEncoder encoder = new HtmlEncoder(UnicodeBlocks.Latin1Supplement, UnicodeBlocks.MiscellaneousSymbols);
+
+            // Act & assert
+            Assert.Equal("&#x61;", encoder.HtmlEncode("a"));
+            Assert.Equal("\u00E9", encoder.HtmlEncode("\u00E9" /* LATIN SMALL LETTER E WITH ACUTE */));
+            Assert.Equal("\u2601", encoder.HtmlEncode("\u2601" /* CLOUD */));
+        }
+
+        [Fact]
+        public void Ctor_WithNoParameters_DefaultsToBasicLatin()
+        {
+            // Arrange
+            HtmlEncoder encoder = new HtmlEncoder();
+
+            // Act & assert
+            Assert.Equal("a", encoder.HtmlEncode("a"));
+            Assert.Equal("&#xE9;", encoder.HtmlEncode("\u00E9" /* LATIN SMALL LETTER E WITH ACUTE */));
+            Assert.Equal("&#x2601;", encoder.HtmlEncode("\u2601" /* CLOUD */));
+        }
+
+        [Fact]
+        public void Default_EquivalentToBasicLatin()
+        {
+            // Arrange
+            HtmlEncoder controlEncoder = new HtmlEncoder(UnicodeBlocks.BasicLatin);
+            HtmlEncoder testEncoder = HtmlEncoder.Default;
+
+            // Act & assert
+            for (int i = 0; i <= Char.MaxValue; i++)
+            {
+                if (!IsSurrogateCodePoint(i))
+                {
+                    string input = new String((char)i, 1);
+                    Assert.Equal(controlEncoder.HtmlEncode(input), testEncoder.HtmlEncode(input));
+                }
+            }
+        }
+
+        [Fact]
+        public void Default_ReturnsSingletonInstance()
+        {
+            // Act
+            HtmlEncoder encoder1 = HtmlEncoder.Default;
+            HtmlEncoder encoder2 = HtmlEncoder.Default;
+
+            // Assert
+            Assert.Same(encoder1, encoder2);
+        }
+
+        [Theory]
+        [InlineData("<", "&lt;")]
+        [InlineData(">", "&gt;")]
+        [InlineData("&", "&amp;")]
+        [InlineData("'", "&#x27;")]
+        [InlineData("\"", "&quot;")]
+        [InlineData("+", "&#x2B;")]
+        public void HtmlEncode_AllRangesAllowed_StillEncodesForbiddenChars_Simple(string input, string expected)
+        {
+            // Arrange
+            HtmlEncoder encoder = new HtmlEncoder(UnicodeBlocks.All);
+
+            // Act
+            string retVal = encoder.HtmlEncode(input);
+
+            // Assert
+            Assert.Equal(expected, retVal);
+        }
+
+        [Fact]
+        public void HtmlEncode_AllRangesAllowed_StillEncodesForbiddenChars_Extended()
+        {
+            // Arrange
+            HtmlEncoder encoder = new HtmlEncoder(UnicodeBlocks.All);
+
+            // Act & assert - BMP chars
+            for (int i = 0; i <= 0xFFFF; i++)
+            {
```

Comment:
I'm admittedly a bit too lazy to look at the product code, but is this unit test just replicating the implementation behavior of the runtime?